### PR TITLE
Add deviceScope property to Device object (from preview to master)

### PR DESCRIPTION
**Pull request by abhipsaMisra**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

## Description of the changes

Add deviceScopes property to Device object.
This field stores an identifier to other edge devices, with whom the device can connect and communicate with even in the absence of Azure IoT Hub connectivity.
https://azure.microsoft.com/en-us/blog/extended-offline-operation-with-azure-iot-edge/

### CODE REVIEW

**Comment by varunpuranik** [comment]
@abhipsaMisra Looks good, but if you cherry pick the commit, we won't lose history of the change.

**Comment by abhipsaMisra** [comment]
I cherry-picked the commits from Azure:preview; so we should be able to preserve the history.

**Comment by varunpuranik** [comment]
Oh okay.. cool

File: e2e/test/RegistryManagerTests.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            Device receivedDevice = await registryManager.AddDeviceAsync(leafDevice).ConfigureAwait(false);
+
+            Assert.IsNotNull(receivedDevice);
+            Assert.AreEqual(leafDevice.Id, receivedDevice.Id);
+            Assert.AreEqual(leafDevice.Scope, receivedDevice.Scope);
+            await registryManager.RemoveDeviceAsync(leafDevice.Id).ConfigureAwait(false);
+            await registryManager.RemoveDeviceAsync(edgeDevice.Id).ConfigureAwait(false);
+        }
+    }
+}
```
Comment (CIPop) [inline e2e/test/RegistryManagerTests.cs]: Nit: NL at EOF.

**Comment by timtay-microsoft** [comment]
When I run this new e2e test, the security scope is always null. Is this the expected behavior for the test @varunpuranik 

I was expecting the scope field to have a non-null value at some point in this test

**Comment by abhipsaMisra** [comment]
Good catch @timtay-microsoft . Confirmed with @varunpuranik that we need to update the api version to "2019-03-30"

**Comment by CIPop** [comment]
@abhipsaMisra @varunpuranik Marking as blocked per @anthonyvercolano who got customer reports saying this doesn't work on a particular scale unit.
/cc @prmathur-microsoft @tameraw

**Pull request merged**